modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/net/http/GetFormParams.java

```diff
@@ -0,0 +1,100 @@
+import org.ballerinalang.model.util.MessageUtils;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BMessage;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.nativeimpl.lang.utils.Constants;
+import org.ballerinalang.natives.AbstractNativeFunction;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.Attribute;
+import org.ballerinalang.natives.annotations.BallerinaAnnotation;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+import org.ballerinalang.runtime.message.StringDataSource;
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+
+/**
+ * Get the Form params from HTTP the Message and return as map.
+ *
+ * @since 0.93
+ */
+@BallerinaFunction(
+        packageName = "ballerina.net.http",
+        functionName = "getFormParams",
+        args = {@Argument(name = "m", type = TypeEnum.MESSAGE)},
+        returnType = {@ReturnType(type = TypeEnum.MAP, elementType = TypeEnum.STRING)},
+        isPublic = true
+)
+@BallerinaAnnotation(annotationName = "Description", attributes = {@Attribute(name = "value",
+        value = "Gets formParam map from HTTP message") })
+@BallerinaAnnotation(annotationName = "Param", attributes = {@Attribute(name = "m",
+        value = "The message object") })
+@BallerinaAnnotation(annotationName = "Return", attributes = {@Attribute(name = "map",
+        value = "The map of form params") })
+public class GetFormParams extends AbstractNativeFunction {
+    @Override
+    public BValue[] execute(Context context) {
+        try {
+            BMessage msg = (BMessage) getRefArgument(context, 0);
+            String contentType = msg.getHeader(Constants.CONTENT_TYPE);
+            if (contentType != null && contentType.contains(Constants.APPLICATION_FORM)) {
+                String payload;
+                if (msg.value().isAlreadyRead()) {
+                    payload = msg.value().getMessageDataSource().getMessageAsString();
+                } else {
+                    payload = MessageUtils.getStringFromInputStream(msg.value().getInputStream());
+                    StringDataSource stringDataSource = new StringDataSource(payload);
+                    msg.value().setMessageDataSource(stringDataSource);
+                    msg.value().setAlreadyRead(true);
+                }
+                if (!payload.isEmpty()) {
+                    return getBValues(processFormParams(payload));
+                } else {
+                    throw new BallerinaException("empty message payload");
+                }
+            }
+        } catch (Throwable e) {
+            throw new BallerinaException("Error while retrieving form param from message: " + e.getMessage());
+        }
+        return new BValue[0];
```

Comment:
This is wrong. This function is returning a map. It should return a null value or an empty map.